spec/queries/get_change_offer_options_spec.rb

```diff
@@ -0,0 +1,126 @@
+require 'rails_helper'
+
+RSpec.describe GetChangeOfferOptions do
+  include CourseOptionHelpers
+
+  let(:ratifying_provider) { create(:provider) }
+  let(:self_ratified_course) { create(:course, :open_on_apply, provider: ratifying_provider) }
+  let(:externally_ratified_course) { create(:course, :open_on_apply, accredited_provider: ratifying_provider) }
+  let(:provider_user) { create(:provider_user) }
+
+  def service(provider_user, course)
+    GetChangeOfferOptions.new(user: provider_user, current_course: course)
+  end
```

Comment:
```suggestion
  let(:service) { described_class.new(user: provider_user, current_course: course)
```    
Can we use a`let` and context blocks around the examples to setup the required data.

Reply:
No, I would rather avoid context blocks. Every scenario is unique.

Reply:
I mean an individual context around each block, similar to how we do in other specs that require data setup.

Reply:
I tried to introduce a `let(:service)` in the later sections of the spec but it clashes with the mocks. In the `make_decisions_courses` section it's important to re-instantiate, otherwise we need to do a lot of `.reload`.

Reply:
I'm not sure I understand. Using a `let` will create a new instance of `service` for every example it's referenced from, so not sure why reloading would be required.

Reply:
Reloading was required because the specs would start with no `make_decisions` permissions, verify no courses were returned, then set the permissions and try again. I've since changes the specs in line with  @cwrw 's suggestion to separate expectations into their own `it` blocks but I'm trying to avoid using too many contexts.